### PR TITLE
Display notifications correctly

### LINKED ISSUE #261: e2e-tests: Fix test for notifications

### LINKED ISSUE #272: User does not receive notifications as expected when workflowitem is assigned

### LINKED ISSUE #283: api: User does not get notified when projected budgets are updated or deleted
When a user is assigned to a project/subproject, he/she does not get notified if the projected budget is updated or deleted.

### LINKED ISSUE #284: ui: Link to project/subproject is not displayed correctly on notifications page
When the user is allowed to see a project/subproject, the button on the right is deactivated, but should be active and vice versa

### LINKED ISSUE #285: ui: Redirect button for fly-in notifications does not work
When a user receives a fly-in notification and clicks on the magnifier button which should redirect to the project/subproject, the error `Error: not implemented` is displayed.

### PR DESCRIPTION
**Pull request by mathiashoeld**
# Changes
## Notification page
The notification page now works as expected. The user can click on the next page and see the next set of notifications. 
Additionally, the magnifier button is now enabled when the user has permissions to see the project/subproject and disabled otherwise. 

## E2E-Test
The e2e tests for notifications have been updated and are working now. 

## Projected Budgets
The assignee of a project/subproject now gets notified if a projected budget was updated or deleted.

## Fly-in notifications
The magnifier button in fly-in notifications now correctly redirects the user to the given project/subproject instead of throwing an error. 

## Notifications
When a user is assigned to a workflowitem, a notification is sent. 

## Issues closed
Closes #283 
Closes #285 
Closes #261 
Closes #284 
Closes #272

### CODE REVIEW

File: api/src/service/domain/workflow/subproject_projected_budget_delete.ts [hunk 1]
```diff
@@ -62,8 +66,25 @@ export async function deleteProjectedBudget(
   if (isEqual(subproject.projectedBudgets, result.projectedBudgets)) {
     return { newEvents: [], projectedBudgets: result.projectedBudgets };
   } else {
+    // Create notification events:
+    const recipients = subproject.assignee
```
Comment (gonzochic) [inline api/src/service/domain/workflow/subproject_projected_budget_delete.ts]: Why is this differently implemented than in `api/src/service/domain/workflow/project_projected_budget_delete.ts`

File: frontend/src/pages/Notifications/NotificationList.js [hunk 2]
```diff
@@ -100,7 +100,7 @@ const NotificationsList = props => {
               markPageAsRead(markMultipleNotificationsAsRead, notifications, notificationOffset, notificationsPerPage)
             }
             color="primary"
-            className={classes.button}
+            className={classes.button + " mark-all-notifications-as-read"}
```
Comment (gonzochic) [inline frontend/src/pages/Notifications/NotificationList.js]: Aren't we using inline-styles instead of classes?

Comment (mathiashoeld) [reply frontend/src/pages/Notifications/NotificationList.js]: yes, you are right. changed it to use only the `data-test` tag in the e2e tests for notifications.

File: api/src/service/domain/workflow/subproject_projected_budget_delete.ts (hunk 1 above)
Comment (mathiashoeld) [reply api/src/service/domain/workflow/subproject_projected_budget_delete.ts]: used same implementation for `project_projected_budget_delete`, `project_projected_budget_update`, `subproject_projected_budget_delete` and `subproject_projected_budget_update`

File: api/src/service/domain/workflow/subproject_projected_budget_delete.spec.ts [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+    assert.instanceOf(result, NotAuthorized);
+  });
+
+  it("The root user doesn't need permission to update a projected budget", async () => {
+    const result = await deleteProjectedBudget(
+      ctx,
+      root,
+      projectId,
+      subprojectId,
+      "Othertestcorp",
```
Comment (gonzochic) [inline api/src/service/domain/workflow/subproject_projected_budget_delete.spec.ts]: The id's don't match with the existing project budget

File: api/src/service/domain/workflow/subproject_projected_budget_delete.spec.ts [hunk 4]
```diff
@@ -0,0 +1,274 @@
+import { assert } from "chai";
```
Comment (gonzochic) [inline api/src/service/domain/workflow/subproject_projected_budget_delete.spec.ts]: Please add the following tests:

- Delete existing PB and check if it is indeed deleted
- Delete non-exsiting PB and check that no error occurs

File: api/src/service/domain/workflow/subproject_projected_budget_delete.spec.ts [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+    if (identity === "bob") return ["bob"];
+    if (identity === "charlie") return ["charlie"];
+    if (identity === "alice_and_bob") return ["alice", "bob"];
+    if (identity === "alice_and_bob_and_charlie") return ["alice", "bob", "charlie"];
+    if (identity === "root") return ["root"];
+    throw Error(`unexpected identity: ${identity}`);
+  },
+};
+
+describe("Update subproject projected budget: permissions", () => {
```
[the remark this answers is not in the document]
Comment (mathiashoeld) [reply api/src/service/domain/workflow/subproject_projected_budget_delete.spec.ts]: changed in the file

File: api/src/service/domain/workflow/subproject_projected_budget_delete.spec.ts (hunk 3 above)
Comment (mathiashoeld) [reply api/src/service/domain/workflow/subproject_projected_budget_delete.spec.ts]: Fixed

File: api/src/service/domain/workflow/subproject_projected_budget_delete.spec.ts (hunk 4 above)
Comment (mathiashoeld) [reply api/src/service/domain/workflow/subproject_projected_budget_delete.spec.ts]: Done
also added unit test for updating PBs (project and subproject)

**Pull request merged**